### DISCUSSION TITLE
Using algorythm only for "overheating" possible?

### DISCUSSION THREAD

**Discussion by nicx** [jmcollin78/versatile_thermostat#368] [Q&A] [answered]
Hey, i hava a heatingpump (Daikin Altherma 3 R) and 12 Shelly TRVs. They are working well in valve mode in general. What I want to achive is, that all valves should stay on 100% as long as the target temperature is not reached exactly. when the target temp is reached, they should either turn off (0%) or decrease smoothly by the algorythm until x degrees over target tempt.
is that possible?

Edit: if you ask why i should do that: its because the heatpump always needs a high water flow, so all valves should always be open. in my case the valve should only close if for example the sun overheats a room.

**Comment by jmcollin78** [comment]
Hello @nicx ,

No this is not possible and this is a very bad idea unless energy is free for you. This is a very bad regulation algorithm with strong overheat.

Why would you do that ? What is the use case ?

**Comment by nicx** [comment]
@jmcollin78 as already mentioned: when using a heat pump this is the most efficient way to go (even regarding energy consumption). the heat pump itself modulates the needed energy. all valves should always be fully open. the only case when a valve should close is external heat source (sun, room fireplace, etc.) to prevent overheating the single room.

**Reply by nicx** [reply]
and in addition: as soon as the flow is obstructed by the valves being turned down, the heat pump cycles, i.e. it switches off and on constantly. this is very bad, so it should be avoided.

**Reply by MBRetrofit** [reply]
You can search for finding a schematic with a boiler in between heat pump and heating system. This way your heating pump will be as close as possible to the most efficient way.

**Comment by jmcollin78** [comment]
I have other opinion about this. Maybe @fredericseiler can expose his opinion about this way of working.

What i sure is that it is not possible to do that now with VTherm.

**Comment by fredericseiler** [comment]
Hi @nicx,

Can you control the heat pump through Home Assistant (setting target temp, on/off, reading ambient temp sensor)?

**Comment by nicx** [comment]
@fredericseiler yes sure. but that doesnt help if one room gets too warm because of direct sun shine or similar... that is why i am looking for a solution not to keep every valve at 100% all the time...

**Comment by fredericseiler** [comment]
Without smart TRVs, you set a target temperature to the main ambient thermostat of the heat pump, you put the TRVs in the same room as the main thermostat at 100% and you keep adjusting the other TRVs all the time because the size of the rooms or radiators are not the same and there also are external factors like you already said. The biggest downside of this setup is that when the target temperature of the main thermostat is reached, the heat pump will heavily slow down the compressor and if you need to warm another room, you just can't. This scenario happens a lot when the thermostat is installed in the same room as a fireplace (a *cool* surprise the first time you go to bed after a nice evening by the fire).

Now, with smart TRVs, the goal is to give each room heating independence while keeping the existing central heating system in place. As previously said in the [other discussion on this topic](https://github.com/jmcollin78/versatile_thermostat/discussions/363), the solution to take advantage of the inverter compressor of nowadays heat pumps, is to let TRVs do their stuff, and apply a target offset on the heat pump based on the "most demanding" TRV, but there are some gotchas I'm still figuring out (negative offset, etc).

About your heat pump cycling when you close some valves, doesn't it have some overpressure protection/bypass in place?

**Comment by nicx** [comment]
this discussion is going a bit in the wrong direction, at least for my original question :D

so I ask again, is there any way with the versatile thermostat Raddon to achieve the following behavior:
- thermostat stays open at 100% until the target temperature is reached
- thermostat slowly closes the valve when the target temperature is exceeded, so that it is completely closed e.g. when the target temperature +1 degree is reached

is this possible or not? if not, I have to look for another solution... ;)

PS: In my opinion a boiler between heatpump and heating system is inefficient, too. keep the system simple, the heat pump always running at lowest modulation, the water flow as high as possible. that's efficient heating ;)

**Reply by jmcollin78** [reply, marked as answer]
So I will answer it again :wink: : no this is not possible with VTherm. You have to do your own way (a simple automation may be).

**Reply by MBRetrofit** [reply]
Strange, the water flow as high is possible together with lowest modulation is equivalent with a very big water volume heated with a very small energy. In most of the cases this insufficient energy will only heat the pipes near the heater exchanger.

The heating pump efficiency is only linked with hourly energy price/solar panels and with outside temperature. 
Usually the requests for heating is never in time sync with the efficient consuming, so a buffer will help a lot to store thermal energy...as you may know that the COP advertised does not exist...

No matter how you pronounce "energy" like - thermal, electrical, nuclear, diesel or wood....is the exactly the same WATT or calorie.

"A calorie is the amount of heat it takes to raise the temperature of 1 gram (0.001 liters) of pure water 1 degree C at sea level."

So COP, in the real world is the algo from Vtherm, as this is the only way to be efficient - to manage the loses not the winning  ( as you bought heating pump because an advertised COP tell's you that the pump will produce more than consume...).
Luckily the energy can't be produced and neither can be consumed, only can be transformed. Always with a ratio less than 1.
